### PR TITLE
add regain package to conda-forge

### PR DESCRIPTION
**Pull request by Vlasovets**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Vlasovets** [comment]
@conda-forge/help-python, hi! 
I would like to publish regain recipe on conda-forge, thank you!

File: recipes/regain/LICENSE [hunk 1]
```diff
@@ -1,4 +1,6 @@
-Copyright (c) 2015-2022, conda-forge
+BSD 3-Clause License
```
Comment (dhirschfeld) [inline recipes/regain/LICENSE:1]: It looks like you've *moved* the top-level `LICENSE.txt`? You shouldn't touch anything outside `recipes/regain` so you'll want to restore that file.

File: recipes/regain/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  license_family: BSD
+  license_file: LICENSE
+  doc_url: https://github.com/fdtomasi/regain
+  dev_url: https://github.com/fdtomasi/regain
+
+extra:
+  recipe-maintainers:
+    - fdtomasi
+    - veronicatozzo
+    - samuelefiorini
```
Comment (dhirschfeld) [inline recipes/regain/meta.yaml:51]: Are these people happy to be maintainers of this `conda-forge` package? Do you not want to add yourself here @Vlasovets?

image

**Comment by dhirschfeld** [comment]
I'm not a maintainer here - just doing a drive-by review. To facilitate this getting merged quickly I think you'll want to address these comments.

File: recipes/regain/meta.yaml (hunk 2 above)
Comment (Vlasovets) [reply recipes/regain/meta.yaml:51]: hi, @dhirschfeld!
and thanks for reviewing my recipe!
I'm in touch with Federici Tomasi (fdtomasi) who has develop this library, and I've just created a recipe to publish it on conda-forge, so I can also publish my own library which depends on this one.
I guess you are right that I should be the maintainer of the recipe, I will change it, thank you!
About the LICENSE.txt, it is exactly the same license agreement from the main [library](https://github.com/fdtomasi/regain/blob/master/LICENSE), what do you mean by restoring it? 
If I remove it from the directory then the test will assert me of not having a licence file.

Comment (dhirschfeld) [reply recipes/regain/meta.yaml]: After the feedstock is created you can reach out and see if anyone else wants to be added as a maintainer and use the bot command to do so:
https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-add-user-username

**Comment by Vlasovets** [comment]
I see, thank you for your help, anyway!

**Comment by Vlasovets** [comment]
> I see, thank you for your help, anyway!

btw, just realised what you mean by restoring LICENSE file, so dumb of me..., already changed it

**Comment by dhirschfeld** [comment]
In this PR you have ***deleted*** the `LICENSE.txt` from the top-level folder.

Compare the top-level folder of your branch with that of `conda-forge`/`main`
<table>
<tr><td>

image

</td><td>

image

</td></tr>
</table>

File: .gitignore [hunk 3]
```diff
@@ -16,3 +16,4 @@ __pycache__
 
 # Jupyter checkpoints
 **/.ipynb_checkpoints/*
+/recipes/gglasso/
```
Comment (dhirschfeld) [inline .gitignore:19]:
```suggestion
```

File: LICENSE.txt [hunk 4]
```diff
@@ -24,4 +24,4 @@ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
 SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
 CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
 OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
Comment (dhirschfeld) [inline LICENSE.txt:27]: Not sure what's going on here (maybe line-endings?) but you'll want to make sure there's no diff.

File: recipes/regain/LICENSE (hunk 1 above)
Comment (dhirschfeld) [reply recipes/regain/LICENSE:1]: It looks like the actual license is included in the sdist therefore you don't need to include a copy here so I'd actually delete the version in this recipe folder.

You only need to include a copy of the license if it's not packaged in the sdist.

File: recipes/regain/meta.yaml (hunk 2 above)
Comment (Vlasovets) [reply recipes/regain/meta.yaml]: @dhirschfeld, thanks again for the valuable feedback!
@conda-forge-admin, may I ask you to take a look at my PR, please?

**Review by dhirschfeld — APPROVED** [review]
@conda-forge/staged-recipes I think this is good to merge!

**Comment by ocefpaf** [comment]
> I'm not a maintainer here - just doing a drive-by review. To facilitate this getting merged quickly I think you'll want to address these comments.

BTW, those "drive-by" are extremely helpful to us! Thanks!!

**Pull request merged**